src/content/language/dot-shorthand.md

```diff
@@ -0,0 +1,328 @@
+---
+title: Dot shorthand syntax
+description: Learn about the shorthand syntax in Dart.
+shortTitle: Dot Shorthands
+prevpage:
+  url: /language/enums
+  title: Enums
+nextpage:
+  url: /language/extension-methods
+  title: Extension methods
+---
+
+:::experimental
+Support for dot shorthand syntax is experimental and
+can only be used on the `main` channel.
+:::
+
+Write more concise Dart code by using the dot shorthand syntax `.foo` 
+instead of `ContextType.foo`. The compiler can infer the type 
+whenever it's clear from the context, letting you access 
+enum values, static members, or constructors more cleanly.
+
+Here is a quick look:
+
+<?code-excerpt "language/lib/classes/shorthand.dart (intro)"?>
+```dart
+// Enum example
+enum Status { none, running, stopped, paused }
+Status currentStatus = .running; // Instead of Status.running
+
+// Static method example
+int port = .parse('8080'); // Instead of int.parse('8080')
+
+// Constructor example
+class Point {
+  final int x, y;
+  Point(this.x, this.y);
+  Point.origin() : x = 0, y = 0;
+}
+Point origin = .origin(); // Instead of Point.origin()
+```
+
+## The role of context type 
+Dot shorthand syntax works because the compiler understands 
+the context type. This is the type it expects an expression 
+to have based on its location. For example, 
+in `Status currentStatus = .running`, the compiler knows 
+a `Status` is expected, so it infers `.running` to mean 
+`Status.running`.
+
+## Using dot shorthand syntax
+
+### Enums
+A primary and highly recommended use case for dot shorthands 
+is with enums, especially in assignments and switch statements, 
+where the enum type is very obvious.
+
+<?code-excerpt "language/lib/classes/shorthand.dart (enums)"?>
+```dart
+// Enum example
+enum LogLevel { debug, info, warning, error }
+
+// Function to get a color code based on log level
+String colorCode(LogLevel level) {
+  return switch (level) {
+    .debug   => 'gray',    // Instead of LogLevel.debug
+    .info    => 'blue',    // Instead of LogLevel.info
+    .warning => 'orange',  // Instead of LogLevel.warning
+    .error   => 'red',     // Instead of LogLevel.error
+  };
+}
+
+// Example usage:
+var warnColor = colorCode(.warning); // Returns 'orange'
+```
+
+### Named constructors
+Dot shorthands are useful for invoking named constructors, 
+static factory methods, or constructors of generic classes 
```

Comment:
I don't think "static factory methods" exist. Factory constructors are probably what you're looking for and you'd probably want to change the `Point.fromList` to be a factory constructor to demonstrate that. And then update the comments in the code snippet.